packages/forms/src/UIForm/UIForm.component.js

```diff
@@ -43,7 +46,14 @@ export default class UIForm extends React.Component {
 			return;
 		}
 		if (Object.keys(jsonSchema).length) {
-			this.setState(merge(jsonSchema, uiSchema));
+			const merged = merge(jsonSchema, uiSchema);
```

Comment:
do we really need to leverage state for this, when its value are allways derived from props and not from a side effect like user input etc ?

Reply:
Yes because merge can be heavy depending on the uiSpec. Executing that at each render can result in perf issues. 
Here we calculate this only if the uiSpec changes